service/src/main/java/bio/terra/profile/service/iam/SamAuthenticatedUserRequestFactory.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.profile.service.iam;
+
+import bio.terra.common.exception.UnauthorizedException;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
+import bio.terra.common.iam.BearerTokenParser;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * An {@link AuthenticatedUserRequestFactory} which always resolves the user email and subjectId
+ * from Sam given a request token.
+ *
+ * <p>This is important for calls made by pet service accounts, which will have a pet email in the
+ * request header, but Sam will return the owner's email.
+ */
+@Component
+public class SamAuthenticatedUserRequestFactory implements AuthenticatedUserRequestFactory {
```

Comment:
Always resolve the incoming token with Sam. WSM does similar, but a bit downstream of the `AuthenticatedUserRequestFactory`. I thought it was useful to do here though because:
1. We always handle requests from pets -- no need to remember to lookup the pet later
2. The service works when not behind an apache proxy (e.g. running locally)